distributor.go

```diff
@@ -53,8 +53,8 @@ type Distributor struct {
 type ReadRing interface {
 	prometheus.Collector
 
-	Get(key uint32) (ring.IngesterDesc, error)
-	GetAll() []ring.IngesterDesc
+	Get(key uint32, n int, heartbeatTimeout time.Duration) ([]ring.IngesterDesc, error)
+	GetAll(heartbeatTimeout time.Duration) []ring.IngesterDesc
```

Comment:
From discussion: consider making the heartbeat timeout a property of the ring instead of passing it in on each call. Then the ring could also report in its metrics `Collect()` method how many total vs. timed out ingesters there are.